tools/clang/test/DXC/dot4add_i8_u8_packed-types.hlsl

```diff
@@ -0,0 +1,21 @@
+// RUN: %dxc /Od /T cs_6_8 %s | FileCheck %s
+
+// Compiling this HLSL would fail this assertion in TranslateDot4AddPacked:
+//
+//     DXASSERT(
+//         !accTy->isVectorTy() && accTy->isIntegerTy(32),
+//         "otherwise, unexpected vector support in high level intrinsic template");
+//
+// Bug was fixed by changing the declarations of dot4add_i8packed and
+// dot4add_u8packed in utils/hct/gen_intrin_main.txt to simply write
+// out their argument and return types, rather than using the $typeN
+// reference syntax.
+
+// CHECK: define void @main()
```

Comment:
We really don't want any more tests that just verify that main is present. Given that the optimized version of this escaped the assert because it didn't actually generate any calls, just verifying that the final form of the dot product calls are in the output should be sufficient.